### PR TITLE
new tests for anydata added

### PR DESCRIPTION
**Pull request by AlexisGaldamez**
new tests for anydata in *not_to_be_null* expectation added

### CODE REVIEW

File: test/core/expectations/test_expect_not_to_be_null.pkb [hunk 1]
```diff
@@ -115,6 +149,32 @@ is
...
+        execute immediate anydata_expectation_block(gc_nested_table_name, gc_nested_table_name||'()', 'convertCollection');
+        --Assert
+        ut.expect(anydata.convertCollection(ut3.ut_expectation_processor.get_failed_expectations())).to_be_empty();
+    end;
+
+    procedure initialized_varray
+    is
+    begin
+        --Act
+        execute immediate anydata_expectation_block(gc_varray_name, gc_varray_name||'()', 'convertCollection');
```
Comment (jgebal) [inline test/core/expectations/test_expect_not_to_be_null.pkb]: maybe just pass 'Collection'

File: test/core/expectations/test_expect_not_to_be_null.pkb [hunk 2]
```diff
@@ -115,6 +149,32 @@ is
         ut.expect(anydata.convertCollection(ut3.ut_expectation_processor.get_failed_expectations())).not_to_be_empty();
     end;
 
+    procedure initialized_object
+    is
+    begin
+        --Act
+        execute immediate anydata_expectation_block(gc_object_name, gc_object_name||'(1)', 'convertObject');
```
Comment (jgebal) [inline test/core/expectations/test_expect_not_to_be_null.pkb]: maybe just pass 'Object'

File: test/core/expectations/test_expect_not_to_be_null.pkb [hunk 3]
```diff
@@ -1,11 +1,45 @@
...
+    is
+        pragma autonomous_transaction;
+    begin
+        execute immediate 'drop type '||gc_object_name;
+        execute immediate 'drop type '||gc_nested_table_name;
+        execute immediate 'drop type '||gc_varray_name;
+    end;
+
+    function anydata_expectation_block(a_object_name IN VARCHAR2, a_object_value IN VARCHAR,
+                                        a_convert_type IN VARCHAR)
```
Comment (jgebal) [inline test/core/expectations/test_expect_not_to_be_null.pkb]: a_object_type varchar2

File: test/core/expectations/test_expect_not_to_be_null.pkb [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
+    procedure drop_types
+    is
+        pragma autonomous_transaction;
+    begin
+        execute immediate 'drop type '||gc_object_name;
+        execute immediate 'drop type '||gc_nested_table_name;
+        execute immediate 'drop type '||gc_varray_name;
+    end;
+
+    function anydata_expectation_block(a_object_name IN VARCHAR2, a_object_value IN VARCHAR,
```
Comment (jgebal) [inline test/core/expectations/test_expect_not_to_be_null.pkb]: We use **all lowercase** formatting.
We use varchar**2**

File: test/core/expectations/test_expect_not_to_be_null.pkb [hunk 5]
```diff
@@ -195,5 +255,40 @@ is
         ut.expect(anydata.convertCollection(ut3.ut_expectation_processor.get_failed_expectations())).not_to_be_empty();
     end;
 
+    procedure null_anydata
+    is
+    begin
+        --Act
+        execute immediate expectations_helpers.unary_expectation_block('not_to_be_null', 'anydata', 'null');
+        --Assert
+        ut.expect(anydata.convertCollection(ut3.ut_expectation_processor.get_failed_expectations())).not_to_be_empty();
```
Comment (jgebal) [inline test/core/expectations/test_expect_not_to_be_null.pkb]: I think we should change that behavior and throw an exception when NULL anydata is passed.

**Comment by jgebal** [comment]
I really like the work you're doing with the tests.
I've added some minor comments on formatting.

**Comment by AlexisGaldamez** [comment]
Thank you, I'll make the changes

**Pull request merged**